### PR TITLE
documenting the api, found at server/routes/README

### PR DESCRIPTION
**Pull request by StevieSong**
#38 ?

### CODE REVIEW

File: server/routes/README.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+Body parameters should be passed in the body of the request, as form data.
+
+**A user must be logged in to use any of these routes**, except for user create.  
+Anytime the current user is referenced, we mean the user that is logged in.
+
+# To Do
+
+* Remove password from user objects
+* Lock down what time format to use in query and body parameters
+* Who can add people to groups? admin only or any group member? configurable?
```
Comment (emilycliu1) [inline server/routes/README.md]: Re: adding people to groups, if the group is public, can anybody add themselves?

File: server/routes/README.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+**Response**
+
+```
+{
+    "groupName": "MyGroup",
+    "admin": "foobar",
+    "description": "A group for friends to track their music progress",
+    "visible": true,
+    "members": [
+        "foobar",
```
Comment (StevieSong) [inline server/routes/README.md]: this is now `{ "username": "foobar" }` per entry in members

File: server/routes/README.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+```
+
+### Edit Group
+To edit something about a group as the admin
+
+    POST /api/group/{groupName}/edit
+
+**Body Parameters**
+
+* key - the name of the attribute to change, either `groupName | admin | description | visible`
```
Comment (StevieSong) [inline server/routes/README.md]: admin --> adminUsername

**Comment by waymao** [comment]
Also, we need to specify the format of the JSON to send if the user is not logged in.
For now I'm doing something like this:
```
{
  "failure": "you are not logged in"
}
```
Is that ok?

File: server/routes/README.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
+# API Routes
+An exhaustive guide to all the API routes and what to send/expect from them.  
+Path parameters are denoted by curly braces `{ }`.  
+Query parameters are denoted by angle brackets `< >`.  
+Body parameters should be passed in the body of the request, as form data.
+
+**A user must be logged in to use any of these routes**, except for user create.
```
Comment (waymao) [inline server/routes/README.md]: We need to specify the format of the JSON to send along with the http error code (403) if the user is not logged in.
For now I'm doing something like this:
```
{
  "failure": "you are not logged in"
}
```
Is that ok?

**Comment by rebecca-millerr** [comment]
Looks like we can access an endpoint to groups the user is a part of but we have no way of knowing what groups the user can access. Seems like we need a column for users including groups they're in (lol did I use those words right I'm not a database person).

File: server/routes/README.md (hunk 1 above)
Comment (StevieSong) [reply server/routes/README.md]: @annelau99

File: server/routes/README.md (hunk 4 above)
Comment (StevieSong) [reply server/routes/README.md]: use 401 for not logged in. 403 is if you're logged in but not allowed to see the resource (e.g. a logged in user trying to access a group they're not a part of). see this [stackoverflow](https://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses)

as far as json, failure --> err? thoughts?

File: server/routes/README.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+
+    POST /api/recording/create
+
+**Body Parameters**
+
+* numPitches - an integer
+* instrument - a string
+* description - a string
+* start - a timestamp
+* end - a timestamp
```
Comment (emilycliu1) [inline server/routes/README.md]: These are going to have to be named startTime and endTime, I believe 'start' is a special keyword in the request body

**Pull request merged**